app/tests/test_generic_api.py

```diff
@@ -402,3 +402,80 @@ def test_head_cache_header(self):
                 self.assertAlmostEqual((expires - now).total_seconds(),
                                        timedelta(seconds=3600).total_seconds(),
                                        delta=2)
+
+
+class ApiCORSHeaderTestCase(StacBaseTestCase):
+
+    @classmethod
+    @mock_s3_asset_file
+    def setUpTestData(cls):
+        cls.factory = Factory()
+        cls.collection = cls.factory.create_collection_sample(
+            name='collection-1',
+            db_create=True,
+        )
+        cls.item = cls.factory.create_item_sample(
+            collection=cls.collection.model,
+            name='item-1',
+            db_create=True,
+        )
+        cls.asset = cls.factory.create_asset_sample(
+            item=cls.item.model,
+            name='asset-1',
+            db_create=True,
+        )
+
+    def test_get_cors_header(self):
+        for endpoint in [
+            '',  # landing page
+            'conformance',
+            f'collections/{self.collection["name"]}',
+            f'collections/{self.collection["name"]}/items/{self.item["name"]}',
+            f'collections/{self.collection["name"]}/items/{self.item["name"]}'
+            f'/assets/{self.asset["name"]}'
```

Comment:
The following endpoint are missing:
- api/stac/v0.9/collections
- api/stac/v0.9/.../items
- api/stac/v0.9/.../assets

Reply:
Well, it's a middleware which treats all endpoints except /search equally, so imho it's safe to not include every single endpoint here. But I can add them if you prefer.

Reply:
I don't mind leave it as is, but it is just in case if we would change the middleware logic it could be useful to have it then in unittest.